src/libostree/ostree-fetcher-soup.c

```diff
@@ -1040,21 +1040,20 @@ on_request_sent (GObject        *object,
                  gpointer        user_data) 
 {
   GTask *task = G_TASK (user_data);
-  OstreeFetcherPendingURI *pending;
-  GCancellable *cancellable;
+  /* Hold a ref to the pending across this function, since we remove
+   * it from the hash early in some cases, not in others. */
+  OstreeFetcherPendingURI *pending = pending_uri_ref (g_task_get_task_data (task));
+  GCancellable *cancellable = g_task_get_cancellable (task);
   GError *local_error = NULL;
   glnx_unref_object SoupMessage *msg = NULL;
 
-  pending = g_task_get_task_data (task);
-  cancellable = g_task_get_cancellable (task);
-
   pending->state = OSTREE_FETCHER_STATE_COMPLETE;
   pending->request_body = soup_request_send_finish ((SoupRequest*) object,
                                                    result, &local_error);
 
   if (!pending->request_body)
```

Comment:
Hmm, just trying to understand. Under what circumstances do we call `remove_pending()` twice? The one in the `out` path is only executed if there was an error, right? And we check for that here by checking for the body. (Or does libsoup also set `local_error` for `REQUESTED_RANGE_NOT_SATISFIABLE`?)